s3/pkg/meta/db/drivers/tidb/multipart.go

```diff
@@ -0,0 +1,240 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package tidbclient
+
+import (
+	"database/sql"
+	"encoding/json"
+	"strings"
+	"time"
+
+	. "github.com/opensds/multi-cloud/s3/error"
+	"github.com/opensds/multi-cloud/s3/pkg/helper"
+	. "github.com/opensds/multi-cloud/s3/pkg/meta/types"
+	pb "github.com/opensds/multi-cloud/s3/proto"
+	log "github.com/sirupsen/logrus"
+)
+
+func (t *TidbClient) CreateMultipart(multipart Multipart) (err error) {
+	m := multipart.Metadata
+	uploadTime := multipart.InitialTime.Format(TIME_LAYOUT_TIDB)
+	acl, _ := json.Marshal(m.Acl)
+	attrs, _ := json.Marshal(m.Attrs)
+	sqltext := "insert into multiparts(bucketname,objectname,uploadid,uploadtime,initiatorid,tenantid,userid,contenttype, " +
+		"location,acl,attrs,objectid,storageMeta,storageclass) " +
+		"values(?,?,?,?,?,?,?,?,?,?,?,?,?,?)"
+	_, err = t.Client.Exec(sqltext, multipart.BucketName, multipart.ObjectKey, multipart.UploadId,uploadTime,
+		m.InitiatorId, m.TenantId, m.UserId, m.ContentType,
+		m.Location, acl, attrs, multipart.ObjectId, multipart.StorageMeta, m.StorageClass)
+	return
+}
+
+func (t *TidbClient) GetMultipart(bucketName, objectName, uploadId string) (multipart Multipart, err error) {
+	sqltext := "select bucketname,objectname,uploadid,uploadtime,initiatorid,tenantid,userid,contenttype," +
+		"location,acl,attrs,objectid,storageMeta,storageclass from multiparts " +
+		"where bucketname=? and objectname=? and uploadid=?;"
+	var acl, attrs, uploadTime string
+	err = t.Client.QueryRow(sqltext, bucketName, objectName, uploadId).Scan(
+		&multipart.BucketName,
+		&multipart.ObjectKey,
+		&multipart.UploadId,
+		&uploadTime,
+		&multipart.Metadata.InitiatorId,
+		&multipart.Metadata.TenantId,
+		&multipart.Metadata.UserId,
+		&multipart.Metadata.ContentType,
+		&multipart.Metadata.Location,
+		&acl,
+		&attrs,
+		&multipart.ObjectId,
+		&multipart.StorageMeta,
+		&multipart.Metadata.StorageClass,
+	)
+	if err != nil && err == sql.ErrNoRows {
+		err = ErrNoSuchUpload
+		return
+	} else if err != nil {
+		return
+	}
+
+	ut, err := time.Parse(TIME_LAYOUT_TIDB, uploadTime)
+	if err != nil {
+		err = handleDBError(err)
+		return
+	}
+	multipart.InitialTime = ut
+
+	err = json.Unmarshal([]byte(acl), &multipart.Metadata.Acl)
+	if err != nil {
+		return
+	}
+
+	err = json.Unmarshal([]byte(attrs), &multipart.Metadata.Attrs)
+	if err != nil {
+		return
+	}
+
+	return
+}
+
+func (t *TidbClient) DeleteMultipart(multipart *Multipart, tx interface{}) (err error) {
+	var sqlTx *sql.Tx
+	if tx == nil {
+		tx, err = t.Client.Begin()
+		defer func() {
+			if err == nil {
+				err = sqlTx.Commit()
+			}
+			if err != nil {
+				sqlTx.Rollback()
+			}
+		}()
+	}
+	sqlTx, _ = tx.(*sql.Tx)
+
+	sqltext := "delete from multiparts where bucketname=? and objectname=? and uploadid=?;"
+	_, err = sqlTx.Exec(sqltext, multipart.BucketName, multipart.ObjectKey, multipart.UploadId)
+	if err != nil {
```

Comment:
Please add error handling

Reply:
@nguptaopensds  thanks for your reviewing,  the error will be returned to upper caller, and the upper caller will check the error.